### PR TITLE
Assigning demployee 2

### PR DESCRIPTION
**Pull request by Joseph-Magee**
to meet the requirements of point 5 assigning  delivery employees to a project.

### CODE REVIEW

File: AssigningDemplyeesToProjects2.sql [hunk 1]
```diff
@@ -0,0 +1,49 @@
+USE JCAC_CormacF
+DELIMITER $$
+DROP PROCEDURE IF EXISTS assign_demploye_to_project $$
+CREATE PROCEDURE assign_demploye_to_project()
```
Comment (jkelso662) [inline AssigningDemplyeesToProjects2.sql]: Procedure names apparently aren't consistent. Although great name! Nice and clear :)

File: AssigningDemplyeesToProjects2.sql [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+BEGIN
+  START TRANSACTION;
+
+CREATE TABLE demployee_project (
+demployee_projectID int PRIMARY KEY AUTO_INCREMENT,
+DemployeeID int,
+ProjectID int);
+
+#drop table demployee_project;
```
Comment (jkelso662) [inline AssigningDemplyeesToProjects2.sql]: Best not to include commented-out code in your main branch :)

File: AssigningDemplyeesToProjects2.sql [hunk 3]
```diff
@@ -0,0 +1,49 @@
...
+INNER JOIN Delivery_Employee
+ON demployee_project.DemployeeID  = Delivery_Employee.DemployeeID
+GROUP BY `project`;
+
+-- check the number of affected rows
+GET DIAGNOSTICS @rows = ROW_COUNT;
+IF @rows = 0 THEN
+  -- if an error occurred,
+  ROLLBACK;
+  SELECT 'Transaction rolled back due to error.';
```
Comment (jkelso662) [inline AssigningDemplyeesToProjects2.sql]: Tabulation 🚨

File: CreateDeliveryEmployeetable2.sql [hunk 4]
```diff
@@ -0,0 +1,41 @@
...
+  NationalInsurance varchar(11));
+
+
+INSERT INTO Delivery_Employee (FirstName, LastName, Salary, BankAccount, NationalInsurance )
+  VALUES ('Joseph', 'Magee', 19000 ,'87212667578643', 'AD913336R'),
+    ('Samuel','Buyan',19000,'21785671234598','AR650465N'),('John', 'Smith', 100000, '34422315894589', 'HG124809K'),
+    ('Joshua','Kelso',900000,'21932721875632','KF242148H'),('Harry','Potter' ,12000, '81760910454545','YT877799U'),
+    ('Mario','Mario',50000,'31571073452189', 'ZX941818B'),('Steve','Rodgers',20000,'52818372454532','BG165344K'),
+    ('Clark','Kent', 45000,'27701114874367','XS633662U'),('Kim-Jong','Un', 10000, '28899322343434','FD714955B'),
+    ('Obiwon','Kenobi',400000,'74044710564354','YU617277W');
```
Comment (jkelso662) [inline CreateDeliveryEmployeetable2.sql:24]: Would be better if each new row was it's own line. But not going to block the merge.

File: AssigningDemplyeesToProjects2.sql [hunk 5]
```diff
@@ -0,0 +1,49 @@
...
+
+ALTER TABLE demployee_project
+  ADD CONSTRAINT fk_demployee_project_Delivery_EmployeeID
+  FOREIGN KEY (DemployeeID)
+  REFERENCES Delivery_Employee(DemployeeID);
+
+ALTER TABLE demployee_project
+  ADD CONSTRAINT fk_demployee_project_ProjectID
+		FOREIGN KEY (ProjectID)
+        REFERENCES Project(ProjectID);
```
Comment (jkelso662) [inline AssigningDemplyeesToProjects2.sql]: Tabulation 🚨

**Review by jkelso662 — APPROVED** [review]
Much better, still a few suggestions I'd like you to read but I'll approve this anyway. Good job!

File: AssigningDemplyeesToProjects2.sql [hunk 6]
```diff
@@ -0,0 +1,49 @@
...
+CREATE PROCEDURE AssignDemployeToProject()
+BEGIN
+  START TRANSACTION;
+
+  CREATE TABLE demployee_project (
+    demployee_projectID int PRIMARY KEY AUTO_INCREMENT,
+    DemployeeID int,
+    ProjectID int);
+
+  #drop table demployee_project;
```
Comment (jkelso662) [inline AssigningDemplyeesToProjects2.sql]: Better not to have commented out code in main!

**Review by jkelso662 — CHANGES_REQUESTED** [review]

**Review by jkelso662 — APPROVED** [review]
Looks great! Just watch out for indentation :)

**Review by Criofann — COMMENTED** [review]
Yipee

**Review by Criofann — APPROVED** [review]

**Pull request merged**